src/main/scala/org/scalameter/Reporter.scala

```diff
@@ -2,26 +2,25 @@ package org.scalameter
 
 
 
-import collection._
-import utils.Tree
+import org.scalameter.utils.Tree
 
 
 
-trait Reporter extends Serializable {
-  def report(result: CurveData, persistor: Persistor): Unit
-  def report(results: Tree[CurveData], persistor: Persistor): Boolean
+trait Reporter[T] extends Serializable {
+  def report(result: CurveData[T], persistor: Persistor): Unit
+  def report(results: Tree[CurveData[T]], persistor: Persistor): Boolean
 }
 
 
 object Reporter {
-  object None extends Reporter {
-    def report(result: CurveData, persistor: Persistor) {}
-    def report(results: Tree[CurveData], persistor: Persistor) = true
+  object None extends Reporter[Nothing] {
```

Comment:
Since `Reporter` is not covariant, we should instead have a method that creates a reporter of the desired type:

    def None[T] = new Reporter[T] { ... }